webapp/lib/controller.dart

```diff
@@ -284,33 +233,52 @@ void command(UIAction action, Data actionData) {
       break;
     case UIAction.addConfigurationResponseEntries:
       ConfigurationResponseData data = actionData;
-      addConfigurationResponseEntries(data.parentTag);
+      addConfigurationResponseEntries(data.parentTag, data.language, data.text);
       break;
   }
 }
 
-void showConfigurationView() {
-  view.contentView.configurationView.tagList.renderTagList(tagData.keys.toList());
-  view.contentView.configurationView.tagResponses.renderResponses(tagData.keys.toList().first, tagData.values.toList().first);
+void showConfigurationView([String selectedTag, Map<String, Map<String, List<String>>> filteredTagData]) {
+  Map<String, bool> tags = new Map.fromIterable(tagData.keys.toList(),
```

Comment:
should it be `filteredTagData` instead of `tagData` here?

Reply:
Oh, I think I see what you're doing.

In general, and this function is a good example of that, it would be good to make functions self-sufficient - ie. to have access only to the data that gets passed to it as arguments, rather than drawing on other data that is not a constant variable (such as `tagData`). Could the code that handles `UIAction.goToConfigurator` (line 220) prepare `filteredTagData` and then pass it on to this method?

Reply:
I've prepared the filtered data when we `UIAction.goToConfigurator` is executed.

Reply:
That's good progress, I think we can work this method even further - basically pass it only the data that it needs, which are: the selected tag, the tag list, and the replies for that tag. I suggest you change the method to take these three things as the arguments, and then adjusted the calling code to prepare this data for it, and the code inside to not do any of the filtering etc.

Reply:
Refactored it to take those 2 params